test/topicFilterCopy.spec.ts

```diff
@@ -0,0 +1,186 @@
+    process.env.TOPIC_NAME,
+    process.env.DEFAULT_SUBSCRIPTION_NAME
+  );
+  await removeAllRules(subscriptionClient);
+}
+
+async function afterEachTest(): Promise<void> {
+  await namespace.close();
+}
+
+async function sendOrder(order: any): Promise<void> {
+  const message: SendableMessageInfo = {
+    body: "",
+    correlationId: order.Priority,
+    label: order.Color,
+    userProperties: {
+      color: order.Color,
+      quantity: order.Quantity,
+      priority: order.Priority
+    }
+  };
+
+  await topicClient.send(message);
+}
+
+async function sendOrders(): Promise<void> {
+  const promises = new Array();
+
+  // Now we can start sending orders.
+  promises.push(sendOrder({ Color: "blue", Quantity: 5, Priority: "low" }));
+  promises.push(sendOrder({ Color: "red", Quantity: 10, Priority: "high" }));
+  promises.push(sendOrder({ Color: "yellow", Quantity: 5, Priority: "low" }));
+  promises.push(sendOrder({ Color: "blue", Quantity: 10, Priority: "low" }));
+  promises.push(sendOrder({ Color: "blue", Quantity: 5, Priority: "high" }));
+  promises.push(sendOrder({ Color: "blue", Quantity: 10, Priority: "low" }));
+  promises.push(sendOrder({ Color: "red", Quantity: 5, Priority: "low" }));
+  promises.push(sendOrder({ Color: "red", Quantity: 10, Priority: "low" }));
+  promises.push(sendOrder({ Color: "red", Quantity: 5, Priority: "low" }));
+  promises.push(sendOrder({ Color: "yellow", Quantity: 10, Priority: "high" }));
+  promises.push(sendOrder({ Color: "yellow", Quantity: 5, Priority: "low" }));
+  promises.push(sendOrder({ Color: "yellow", Quantity: 10, Priority: "low" }));
+
+  // wait until all the send tasks are complete
+  await Promise.all(promises);
+}
+
+async function receiveOrders(client: SubscriptionClient): Promise<ServiceBusMessage[]> {
+  const receivedMsgs: ServiceBusMessage[] = [];
+  const receiveListener = client.receive(
+    (msg: ServiceBusMessage) => {
+      receivedMsgs.push(msg);
+      return Promise.resolve();
+    },
+    (err: Error) => {
+      should.not.exist(err);
+    }
+  );
+
+  await delay(3000);
+  await receiveListener.stop();
+
+  return receivedMsgs;
+}
+
+describe.only("Boolean filters", function(): void {
+  beforeEach(async () => {
+    await beforeEachTest();
+  });
+
+  afterEach(async () => {
+    await afterEachTest();
+  });
+
+  async function addFileterAndReceiveOrders(
+    bool: boolean,
+    client: SubscriptionClient
+  ): Promise<ServiceBusMessage[]> {
+    await subscriptionClient.addRule("BooleanFilter", bool);
+    const rules = await subscriptionClient.getRules();
+    should.equal(rules.length, 1);
+    should.equal(rules[0].name, "BooleanFilter");
+
+    await sendOrders();
+    const receivedMsgs = await receiveOrders(client);
+
+    return receivedMsgs;
+  }
+
+  it("Subscription with default filter receives all messages", async function(): Promise<void> {
+    const receivedMsgs = await addFileterAndReceiveOrders(false, defaultSubscriptionClient);
```

Comment:
The idea of having a `defaultSubscriptionClient` is that we dont add/remove rules from it.

`addFileterAndReceiveOrders` will add a rule.

Right now, your test is passing because the default rule on the default subscription wasnt removed.

Reply:
Can I remove defaultSubscriptionClient ?

Reply:
I added that because of this test case `A subscription with default filter should receive all messages sent via the topic`.